apps/predictions/lib/stream_worker.ex

```diff
@@ -18,44 +17,38 @@ defmodule Predictions.StreamSupervisor.Worker do
   end
 
   @impl Supervisor
-  def init(key) do
-    sses_stream_name = sses_stream_name(key)
-    api_stream_name = :"predictions_api_stream_#{key}"
-    prediction_stream_name = :"predictions_data_stream_#{key}"
+  @doc "The `filters` argument is a string representing the filter parameters of
+  the resultant API call e.g. 'filter[route]=CR-Foxboro&filter[direction_id]=0'"
+  def init(filters) do
+    sses_stream_name = sses_stream_name(filters)
+    api_stream_name = :"predictions_api_stream_#{filters}"
+    prediction_stream_name = :"predictions_data_stream_#{filters}"
```

Comment:
question(blocking): are these filters validated before they're used here? Since both of these items are creating new atoms, we need to be careful that users/attackers can't generate whichever ones they want.

Reply:
clarification: are we concerned about new attack atoms being part of these names, or getting into the `/predictions?#{filters}` request? (or both for separate reasons?)

Reply:
I'm thinking about whether clients can subscribe to arbitrary filters. Looking back at the code though, I think that the functionality works like this:
- client goes to stops page
- client subscribes to `predictions:stop:<id>`
- server looks up the route patterns for the routes to subscribe to
- stream is created for the given routes

As long as it's not possible for clients to subscribe to an arbitrary client-controlled route, then this should be okay. But it's something to keep an eye on.

Reply:
Yeah, part of StreamTopic's purpose is absolutely to guard against that sorta outcome, and the tests should demonstrate that to an extent. I think it'll be ok, and I can add a few more tests to validate.